### PR TITLE
enhance: add more logs for target updating

### PR DESCRIPTION
**Pull request by yah01**
- add more logs about the condition satisfying

### CODE REVIEW

**Comment by yah01** [comment]
related https://github.com/milvus-io/milvus/issues/29082

File: internal/querycoordv2/observers/target_observer.go [hunk 1]
```diff
@@ -279,12 +284,21 @@ func (ob *TargetObserver) updateNextTargetTimestamp(collectionID int64) {
 }
 
 func (ob *TargetObserver) shouldUpdateCurrentTarget(ctx context.Context, collectionID int64) bool {
+	log := log.Ctx(ctx).WithRateGroup(
+		"qcv2.TargetObserver",
```
Comment (congqixia) [inline internal/querycoordv2/observers/target_observer.go:288]:
```suggestion
		fmt.Sprintf("qcv2.TargetObserver-%d", collectionID),
```

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**